Add priorityClassName support

anxolerd opened the pull request:
## Description
Allow to set priorityClassName for master pod. This will make possible
for jenkins master to either not be preempted on other's pod low
resources or to preempt lesser priority pods on resources shortage.

## API changes
Add `spec.master.priorityClassName` field to `Jenkins` CR, which corresponds to `spec.priorityClassName` field in `Pod` spec. Field type is `string`, it should be either name of a valid `scheduling.k8s.io/v1/PriorityClass` or empty string.

## Linked issues
Closes #321

tomaszsek commented on pkg/controller/jenkins/configuration/base/validate.go:
```diff
@@ -407,3 +414,17 @@ func (r *ReconcileJenkinsBaseConfiguration) validateCustomization(customization
+	var messages []string
+	if len(priorityClassName) > 0 {
+		priorityClass := &schedulingv1.PriorityClass{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: priorityClassName}, priorityClass)
```
One missing part. You have to update https://github.com/jenkinsci/kubernetes-operator/blob/master/deploy/role.yaml and give operator permission to get `PriorityClass`.

anxolerd replied:
Changed that in 88fa77d

tomaszsek commented:
It will be great if we could check this in e2e https://github.com/jenkinsci/kubernetes-operator/blob/master/test/e2e/jenkins.go#L69.

anxolerd commented:
@tomaszsek , I reverted validation, as it compicates things in multitenant environment and added e2e test.

I have a question regarding e2e tests: is it possible to check negative cases (i.e. when reconcile loop fails) in those?

anxolerd commented:
Tests are failing due to this: https://github.com/kubernetes/kubernetes/issues/60596

tomaszsek commented:
You can bump minikube Kubernetes version here https://github.com/jenkinsci/kubernetes-operator/blob/master/config.minikube.env#L3

anxolerd commented:
@tomaszsek , after tons of struggling with e2e, i finally managed to make tests pass!

tomaszsek commented:
@anxolerd Thanks :+1:
